src/components/limited-restoration-panel/LimitedRestorationPanel.vue

```diff
@@ -0,0 +1,104 @@
+<template id="limited-restoration-panel">
+  <v-radio-group class="mt-0 pt-0 ml-8" v-model="selectMonths" mandatory>
+    <v-radio id="twentyFour-radio" name="selectMonths" label="Two Years" :value=24 />
+    <v-radio id="eighteen-radio" name="selectMonths" label="18 months" :value=18 />
+    <v-radio id="twelve-radio" name="selectMonths" label="12 months" :value=12 />
+    <v-radio id="six-radio" name="selectMonths" label="6 months" :value=6 />
+    <v-row class="ml-0">
+      <v-radio id="custom-months" name="selectMonths" label="Month(s): " value="customMonths">
+      </v-radio>
+      <v-text-field class="shrink mx-4" type="number" min=1 max=24 :rules="monthRules" v-model="numberOfMonths"
+        :disabled="!customMonths" @change="onMonthsChanged" filled />
+    </v-row>
+  </v-radio-group>
+</template>
+<script lang="ts">
+
+import { Component, Emit, Mixins, Prop, Watch } from 'vue-property-decorator'
+import { DateMixin } from '@/mixins' // NB: local mixin (StoryBook can't find it otherwise)
+
+@Component({})
+export default class LimitedRestorationPanel extends Mixins(DateMixin) {
+  @Prop({ default: '' }) readonly currentDate!: string
+
+  @Prop({ default: '' }) readonly expiryDate!: string
+
+  // Local properties
+  private customMonths = false
+  private selectMonths = ''
+  private numberOfMonths = 1
+  private monthRules: ((data: string) => void)[] = []
+
+  /**
+   * Called when the component is mounted.
+   * Auto select the previous months input of user (if any).
+   * If there is no previous input, auto select 24 months.
+   */
+  mounted (): void {
+    this.setMonthRules()
+    if (this.expiryDate !== null && this.expiryDate !== undefined) { // Unit tests fail without the first condition.
+      let draftMonths = this.subtractDates(this.currentDate, this.expiryDate)
+      if (draftMonths !== 24 && draftMonths !== 18 && draftMonths !== 12 && draftMonths !== 6) {
+        this.selectMonths = 'customMonths'
+        this.numberOfMonths = draftMonths
+        this.setCustomMonths(this.numberOfMonths)
+      } else {
+        this.selectMonths = draftMonths
+      }
+    } else {
+      this.selectMonths = 24
+    }
+  }
+
+  /** The validation rules for the Month. */
+  private setMonthRules (): void {
+    this.monthRules = [
+      (v:string) => !!v || 'Please enter the number of months',
+      (v:string) => (v > 0 && v <= 24) || 'This field must be between 1 to 24', // Keep month between 1 and 24
+      (v:string) => (v % 1 === 0) || 'Please enter a valid value' // Check if month is an integer
+    ]
+  }
+
+  // Emit the expiry date.
+  @Emit('expiry')
+  private expiryChanged (event:string): string {
```

Comment:
Please run `npm run lint` before you commit further changes. I think it might catch, and maybe fix, the parameter type spacing.

Reply:
I ran lint before committing. Weird that it didn't catch it. I'll add the spacing manually.